static/src/javascripts/projects/common/modules/article/space-filler.js

```diff
@@ -1,66 +1,47 @@
-/* jscs:disable disallowDanglingUnderscores */
 define([
-    'common/utils/$',
-    'common/utils/fastdom-idle',
+    'common/utils/fastdom-promise',
     'Promise',
     'raven',
     'common/modules/article/spacefinder'
 ], function (
-    $,
-    idleFastdom,
+    fastdom,
     Promise,
     raven,
     spacefinder
 ) {
-    return new SpaceFiller();
+    var lastIteration = Promise.resolve(false);
 
-    function SpaceFiller() {
-        var lastInsertion = Promise.resolve(false);
+    /**
+     * A safer way of using spacefinder.
+     * Given a set of spacefinder rules, applies a writer to the first matching paragraph.
+     * Uses fastdom to avoid layout-thrashing, but queues up asynchronous writes to avoid race conditions. We don't
+     * seek a slot for a new component until all the other component writes have finished.
+     *
+     * @param rules - a spacefinder ruleset
+     * @param writer - function, takes a para element and injects a container for the new content synchronously. It should NOT use Fastdom.
+     *
+     * @returns {Promise} - when insertion attempt completed, resolves 'true' if inserted, or 'false' if no space found
+     */
+    function fillSpace(rules, writer) {
```

Comment:
well, yes it does. It's not a "para" we're looking for, but a space to be filled by the writer.